TEDU-52 Revamped the Home page topics (#176)
mautic/mautic-documentation: pages/01.Home/02.Contributing/01.Style Guide/docs.en.md

     * @param bool   $debug       Whether to enable debugging or not
     *
     * @api
     */
    public function __construct($environment, $debug)
    {
        defined('MAUTIC_ENV') or define('MAUTIC_ENV', $environment);
        defined('MAUTIC_VERSION') or define(
            'MAUTIC_VERSION',
            self::MAJOR_VERSION.'.'.self::MINOR_VERSION.'.'.self::PATCH_VERSION.self::EXTRA_VERSION
        );

        parent::__construct($environment, $debug);
    }
    ```
```

</details>
</details>

### Links

[link macro]: <http://example.com>
[testing]: <./../plugins/integration_test.html>

Often you'll want to make a link to another place in the documentation.  We prefer to group links at the bottom of a page, and provide a reference macro in the text.
This make linking to the same place much easier.  In Markdown, it looks like this:

#### Absolute links

You can have different links in your text, some leading [here][link macro] and others leading [there][link macro].

```markdown
You can have different links in your text, some leading [here][link macro] and others leading [there][link macro].

[link macro]: <http://example.com>
```

#### Relative links

> These will link to [`plugins/integration_test.html`][testing] on the documentation website created from the [_`plugins/integration_test.md`_][testing] source file.

Reviewer: This uses the old format, let's update it to avoid confusion!

```suggestion
These will link to [`/plugins/plugin-resources/testing-integrations`][testing] on the documentation website created from the [_`pages/13.Plugins/00.General%20Resources/01.Integration%20test/docs.en.md`_][testing] source file.  These should still use a link macro as with absolute links.
```

> The first line is an example from within the same section.  The second for anywhere else in the directory structure.

Reviewer: ```suggestion
```

We do not use that syntax any more

```markdown

> [testing]: <integration_test.html>

Reviewer: ```suggestion
[testing]: </plugins/plugin-resources/testing-integrations>
```

> [testing]: <./../plugins/integration_test.html>

Reviewer: ```suggestion
```
We do not need this any more, afaik

```

#### Heading anchors

Heading anchors enable linking directly to a Markdown heading from within the same document very easily.  The anchors are auto-generated for all headings.
The link target is specified inline.

```markdown
You can have different anchors in your text, for example to this [Section][#anchor-section] right here.

### Images

> Images should be placed in the media subfolders in the different sections of this repository.

Reviewer: ```suggestion
Images should be placed in the same folder as the page they are used on. The image should be added to Git as a file to be included in the repository, not uploaded to Github and hosted by Github.
```

Link format:

```markdown
![alternative text here](image-reference.png "Tooltip text here")
```

#### Relative image links

To display an image already in the documentation repository, use a relative path.

> The first line is an example of using images in the same section.  The second for anywhere else in the directory structure.

Reviewer: Please check if this is still this case - it was in our old Gitbook docs but I do not know if that is still the case now.

Author: I tried using a relative link to link to an image somewhere else in the directory structure. It didn't work. So, removed this topic.

```markdown
![form rebuild cache](media/rebuild.png "Rebuild form cache"")

![form rebuild cache](./../forms/media/rebuild.png "Rebuild form cache")
```

#### Absolute image links

> For images that cannot be uploaded via the GitHub web interface, upload them to a public URL service and use the generated link.

Reviewer: Do not want this to be happening - all images should be in the repo - let's remove this section (it often results in broken images and is subject to potential abuse)

Author: Removed this topic.

```markdown
![apple](http://example.com/images/apple.png "Like this delicious apple")
```

> **Warning**
>

> > ##### internal absolute image link

Reviewer: Remove this bit too, think it is not relevant.

Author: Removed.
